qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1359,10 +1359,27 @@ def validate(self, restriction_dict):
                         if datatype == datetime:
                             val = str(val)
                             try:
-                                datetime.strptime(val, '%m/%d/%y %H:%M:%S')
+                                datetime.strptime(val, '%m/%d/%Y %H:%M:%S')
```

Comment:
Argh! That hurts! So sad that datetime doesn't support the "or" operator. What about something like this?
```python
formats = ['%m/%d/%Y %H:%M:%S', '%m/%d/%Y %H:%M', '%m/%d/%Y %H', '%m/%d/%Y', '%m/%Y', '%Y']
dt = None
for fmt in formats:
    try:
        dt = datetime.strptime(val, fmt)
    except ValueError:
        pass
if dt is None:
    warning_msg.append(....)
```